app/utils.py

```diff
@@ -0,0 +1,33 @@
+from aries_cloudcontroller import AriesAgentController, AriesTenantController
+from fastapi import Header, HTTPException
+
+import os
+from typing import Optional
+
+admin_url = os.getenv("ACAPY_ADMIN_URL")
+admin_port = os.getenv("ACAPY_ADMIN_PORT")
+is_multitenant = os.getenv("IS_MULTITENANT", False)
+
+
+def create_controller(req_header: Header):
```

Comment:
nice - we want to use this pattern more and maybe even call it a factory (potentially)...

Reply:
Yeah something like this. I reckon even making this a context managable class might work here. (the reason being that we otherwise have to decide to make a breaking change to cloudcontroller and I'm not sure Will wants this in the controller)

Reply:
So the ClientSession (which is what is closed by the terminate) already supports context managers etc...

and furthermore, adding the methods would not break any clients already using the default approach - they wouldn't call the methods.

Reality is it would improve the quality and usability of these Controllers as the clients would no longer need to know to call terminate when they're done.